src/parsing/staticallyCompileXPath.ts

```diff
@@ -81,6 +84,24 @@ export default function staticallyCompileXPath(
 			processProlog(prolog, rootStaticContext);
 		}
 
+		if (compilationOptions.annotateAst) {
+			annotateAst(ast, {
+				staticContext: rootStaticContext,
+				query: xpathString,
+			});
+
+			if (compilationOptions.logUnannotatedQueries) {
+				const [totalNodes, annotatedNodes] = countQueryBodyAnnotations(ast);
+
+				if (totalNodes !== annotatedNodes) {
+					// We are logging an error here so we can easily pipe the list of all failed queries into a file
+					// Example: `npm run test 2> failedQueries.txt`
+					// tslint:disable-next-line:no-console
+					console.error(xpathString);
+				}
```

Comment:
This part is an interesting framework. So, if we ever need more flags, we can introduce them? Not sure if Fonto wants so many flags in their repo. Oh well at least they are optional which won't raise a big problem. I'm just rambling here

Reply:
Yeah we definitely have to check this with Martin but for now this was the best approach I could come up with